core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/ErrorHandler.java

```diff
@@ -0,0 +1,50 @@
+package com.microsoft.applicationinsights.channel.concrete.inprocess;
+
+import org.apache.http.HttpStatus;
+
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandler;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandlerArgs;
+import com.microsoft.applicationinsights.internal.channel.common.TransmissionPolicyManager;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+
+public class ErrorHandler implements TransmissionHandler {
+
+	private TransmissionPolicyManager transmissionPolicyManager;
+	public ErrorHandler(TransmissionPolicyManager policy)
+	{
+		this.transmissionPolicyManager = policy;
+	}
+	
+	@Override
+	public void onTransmissionSent(TransmissionHandlerArgs args) {
+		
+		validateTransmissionAndSend(args);
+	}
+
+	public boolean validateTransmissionAndSend(TransmissionHandlerArgs args) {
```

Comment:
Does this method need to be public? If it is not accessed from outside, reduce the scope to private or default.

Reply:
Ah so mockito version 1.10 does not allow mocking private methods. Is this method a part of interface? That's why I was suggesting since this is used widely we can plug it into the interface and then mock it. If that route is not valid for some reason then we might better use reflection. Reducing the publicly exposed methods will improve our abilities to modify it without affecting anyone. Users can take dependency on public methods and modification of signature can thus be very cumbersome.

Reply:
I definitely want to keep the tests for this method as it checks the actual logic of the handler(s). I'll do a bit of looking around to see what are some good solutions to keep the scope down. Possibly extending the class as a package private and setting those methods to protected.

Reply:
Fixed this by using the package scope. Only classes defined in the packages can see the method.